### DISCUSSION TITLE
Supabase API OR filter with multiple conditions

### DISCUSSION THREAD

**Discussion by kavitang** [supabase/supabase#12114] [Questions] [answered]
Hey,

My goal is to include multiple conditions in the `or` filter. These conditions are not from the same tables. For example, say I'm querying `table1` and another junction table `table1_table2` which has `table1_id` and `table2_id.

`table1` fields `id, name, email`
`table2` fields `id, title, age`
`table1_table2` fields `table1_id`, `table2_id`

I have two filters, `name.eq.ABC` and `table1_table2.table2_id.age.eq.10`. I have tried the following conditions but none of them work

Attempt 1

```
or('name.eq.ABC, table1_table2.table2_id.age.eq.10', { foreignTable: "table1_table2"});
// This results in a payload `table1_table2.or: 'name.eq.ABC, table1_table2.table2_id.age.eq.10'` which throws an error (no name in table1_table2)
```
Attempt 2
```
or('name.eq.ABC')
or('table2_id.age.eq.10', { foreignTable: "table1_table2"});
// This separates the two but then the result is not an OR of these two conditions
```

How do I combine these two? I want all entries with both conditions. Any suggestions would be highly appreciated, thanks!

**Comment by GaryAustin1** [comment, marked as answer]
It is not possible to use .or on different tables currently...
https://github.com/supabase/supabase/discussions/2234

**Reply by kavitang** [reply]
Thank you for the response, I'll see if the rpc method works for my case.
